materials/signet-genesis-node.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# Genesis Node Setup
+#
+# - Deploys a local signet.
+# - Mines 16 blocks as bootstrapping for the SRI pool.
+#
+# Ensure the following environment variables are set:
+#
+# $BITCOIND=$HOME/bitcoin/src/bitcoind
+# $BITCOIN_CLI=$HOME/bitcoin/src/bitcoin-cli
+# $BITCOIN_UTIL=$HOME/bitcoin/src/bitcoin-util
+# $MINER=$HOME/bitcoin/contrib/signet/miner
+# $BITCOIN_DATA_DIR=$HOME/.bitcoin
+#
+
+# Check if the environment variables are set
+if [ -z "$BITCOIND" ] || [ -z "$BITCOIN_CLI" ] || [ -z "$BITCOIN_UTIL" ] || [ -z "$BITCOIN_DATA_DIR" ] || [ -z "$MINER" ]; then
+  echo "Please make sure all environment variables (BITCOIND, BITCOIN_CLI, BITCOIN_UTIL, BITCOIN_DATA_DIR, MINER) are properly set. Exiting."
+  exit 1
+fi
+
+# Check if the specific port is in use
+if lsof -i :38332 | grep -q LISTEN; then
+  echo "Port 38332 is already being used, which means there's probably a bitcoind already running in signet mode. Exiting."
+  exit 1
+fi
+
+echo "This script is about to erase some contents of the provided BITCOIN_DATA_DIR. Do you want to continue? (yes/no)"
+read -r user_input
+
+if [ "$user_input" != "yes" ]; then
+  echo "User did not consent. Exiting."
+  exit 1
+fi
+
+rm -rf $BITCOIN_DATA_DIR/signet
+rm -rf $BITCOIN_DATA_DIR/bitcoin.conf
+
+cat << EOF > $BITCOIN_DATA_DIR/bitcoin.conf
+[signet]
+prune=0
+txindex=1
+server=1
+rpcallowip=0.0.0.0/0
+rpcbind=0.0.0.0
+rpcuser=mempool
+rpcpassword=mempool
+# OP_TRUE
+signetchallenge=51
+EOF
```

Comment:
if we have this as a separate file, it feels redundant to also have this heredoc on the script

given the original goal of this script (deterministic automation of genesis node deployment), I would drop `genesis-bitcoin.conf` in favor of the heredoc